source/includes/library-reference/_simulator.html.md

```diff
@@ -0,0 +1,158 @@
+# Simulator Class
+
+> ![Simulator state image][1]
+
+This class is used to interface with the server while training or running predictions against
+a **BRAIN**. It is an abstract base class, and the developer must create a subclass.
+
+The `Simulator` class has a close relationship with the **Inkling** file associated with
+the **BRAIN**. The name used to construct the `Simulator` must match the name of the simulator
+in the *Inkling* file.
+
+There are two main methods one must override, `episode_start` and `simulate`.
+
+## Brain brain()
+Returns the brain being used for this simulation.
+
+## string name()
+Returns the simulator name that was passed in when constructed
+
+## bool predict()
+Returns weather or not the simulation is setup to run in predict or training mode
+
+## string objective_name()
+Accessor method which returns the name of the current objective.
+Objective may be updated before `episode_start` is called. When running
+for prediction and during start up, objective will return an empty string.
+
+## episode_start(parameters, initial_state)
+
+### Arguments
+- `parameters`     InklingMessage of initialization parameters for an episode as defined in
```

Comment:
And because of the naming conflict with function parameters, `parameters` are often `*args` or 'params'.

Reply:
how about "episode_parameters"? might be a better name.